packages/tsx/tsx.mjs

```diff
@@ -36,19 +36,11 @@ async function resolveTSX(specifier, ctx, nextResolve) {
 		/** @type {FileURL} */ (ctx.parentURL ?? pathToFileURL(path.join(cwd(), 'whatever.ext')).href),
 	);
 
-	if (jsxExts.has(ext)) {
+	if (ext === '.jsx' || ext === '.tsx' || ext === '.mts' || ext === '.ts') {
 		return {
 			...nextResult,
 			// @ts-ignore https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71493
-			format: 'jsx',
-		};
-	}
-
-	if (tsxExts.has(ext)) {
-		return {
-			...nextResult,
-			// @ts-ignore https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71493
-			format: 'tsx',
+			format: '(j|t)sx',
```

Comment:
This looks very weird, like it's supposed to be a regex.

If you want to de-dupe, let's break out format

```js
let format;
if (jsxExts.has(ext)) format = 'jsx';
else if (tsxExts.has(ext)) format = 'tsx';

if (format) {
  return {
    ...nextResult,
	  // @ts-ignore https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71493
    format,
  };
}
```

Reply:
the question is why we need two different forment ?

Reply:
We technically don't (with the current setup). That may change, and keeping them differentiated avoids a breaking change. But also, someone may want to operate on only one or the other.

ps the change currently in this PR is a breaking change.

Reply:
on this point I didn't agree with you. Because format is only internal (I mean loader) thing. Also esbuild don't care if it's `jsx`, `tsx` or idk it transform it. And even the `js` is tempted to be transformed

Reply:
Format is _not_ internal. Any hook in the chain can see it:

```js
module.register('@nodejs-loaders/tsx')
module.register('jsx-something')
```

```js

export async fn resolveJSXSomething(…) {
  const { format } = await nextResolve(…);
  console.log(format); // 'jsx'
}
```

Reply:
I'm okay to keep this behavior to unsure we are not making breaking changes. But should we recommend or not to do that.

Reply:
Hmm? You're saying we should recommend people don't depend on our external API?

Reply:
> Hmm? You're saying we should recommend people don't depend on our external API?

Should we recommend people to use a loader in an other loader ? IMO no the resolution chain have this role.

I'm closing this discution here because it's resolved for this pr. maybe let's open an other issue about that.

Reply:
Sorry, I don't mean to poo-poo your idea.

The hooks chain was specifically designed to be collaborative: each hook can get help from others. That's ½ of why `next` exists.